### PR TITLE
Post blocks

### PR DESCRIPTION
**Pull request by urchykli**
There are quite a few variations of the post block, but I think I got everything.  The desktop styles are [here](https://www.figma.com/file/cAg0A1ww0s2KbJdIbUvl2j/PONI-Redesign-Handoff?node-id=1%3A5099) and the other sizes can be found on the individual archive mockup pages.

You'll need to pull from dev to get the test posts locally.  The blocks can be found at the links below.

 - [NPN](https://localhost:3000/news/)
 - [Events](https://localhost:3000/events/)
 - [Analysis](https://localhost:3000/analysis/)
 - [Projects](https://localhost:3000/projects/)
 - [Programs](https://localhost:3000/programs/)
 - [Program Updates](https://localhost:3000/updates/) this includes the monthly newsletter as well.

### CODE REVIEW

File: wp-content/themes/csisnuclearnetwork/inc/template-tags.php [hunk 1]
```diff
@@ -159,12 +159,19 @@ function nuclearnetwork_posted_on( $date_format = null ) {
...
+	$post_type = get_post_type();
 
 	if  ( $date_format ) {
 		$date = $date_format;
 	}
 
-	echo '<div class="post-meta post-meta__date">' . get_the_time( $date ) . '</div>';
+	if ( $post_type === 'news' ) {
+		echo '<div class="post-meta post-meta__date">' . get_the_time( $date ) . '</div>';
+	} elseif ( $post_type === 'events' || $post_type === 'programs' || $post_type === 'projects' ) {
```
Comment (jnschrag) [inline wp-content/themes/csisnuclearnetwork/inc/template-tags.php:170]:
```suggestion
	} elseif ( in_array( $post_type, array( 'events', 'programs', 'projects') )  ) {
```

If you find yourself checking against multiple post types, it's better to check if they're in an array to simplify your if statements. See more info about [`in_array` here](https://www.php.net/manual/en/function.in-array.php).

File: wp-content/themes/csisnuclearnetwork/inc/template-tags.php [hunk 2]
```diff
@@ -205,11 +212,13 @@ function nuclearnetwork_authors() {
 		$authors = the_author_posts_link();
 	}
 
-	if ( !$authors ) {
+	$post_type = get_post_type();
+
+	if ( !$authors || $post_type === 'events' || $post_type === 'programs' || $post_type === 'projects' ) {
```
Comment (jnschrag) [inline wp-content/themes/csisnuclearnetwork/inc/template-tags.php:217]: See note above.

File: wp-content/themes/csisnuclearnetwork/inc/template-tags.php [hunk 3]
```diff
@@ -377,3 +386,53 @@ function nuclearnetwork_display_footnotes() {
...
+			$post_type_name = 'Program';
+			$tax_name = '';
+			$slug = $post_type;
+		} elseif ($post_type === 'post' ) {
+			$post_type_name = 'Analysis';
+			$tax_name = 'analysis_subtype';
+			$slug = 'analysis';
+		}
+		
+		echo '<div class="post-meta post-meta__terms"><a href="/' . $slug . '" class="post-meta__terms-type text--bold">' . $post_type_name . get_the_term_list( $post->ID, $tax_name, ' / </a>', ', ') . '</div>';
```
Comment (jnschrag) [inline wp-content/themes/csisnuclearnetwork/inc/template-tags.php:421]: I'm concerned about hardcoding these URLs and titles in to the theme like this - makes it a lot harder to change in the future if needed. Is there a way you'd be able to use the built in permalink functions here to simplify this? If not, then be sure to add a note to the README that this is a part of the theme that would need to be manually updated if they ever want to update the language for these content types.

File: wp-content/themes/csisnuclearnetwork/template-parts/block-post.php [hunk 4]
```diff
@@ -11,19 +11,114 @@
...
+			the_excerpt(); 
+			if ( isset($event_info['event_registration_link']) && !empty($event_info['event_registration_link']) && $is_future_event && !$legacy_event_date ) { ?>
+			<a href="<?php echo esc_url( $event_info['event_registration_link'] ); ?>" class="post-block__register btn btn--outline-blue">Register<?php echo nuclearnetwork_get_svg( 'arrow-external' ); ?></a>
+			<?php
+			} ?>
+		</div>
+		<?php
+		if ( $event_start_date || $event_location ) { ?>
+		<div class="post-block__event-meta">
+			<?php
```
Comment (jnschrag) [inline wp-content/themes/csisnuclearnetwork/template-parts/block-post.php:103]: I don't think you need to change these now, but in the future I would recommend making these functions in the `template-tags.php` file just like the functions to display the list of categories, tags, etc. It keeps your functional logic separated from your display logic, and makes it easier to call this information in multiple places if needed.

**Review by jnschrag — CHANGES_REQUESTED** [review]
Nice job on this, Lindsay! There was a lot here and I think you did a great job.

## General Notes
- I should have mentioned this sooner when I saw that the design called for this, but it's actually not ideal performance-wise to toggle box-shadows. Instead, the recommend method for this kind of thing is actually to apply a pseudo-element to the element that has the box-shadow applied, and instead transition the opacity of the pseudo element instead. Check out [this article](https://alligator.io/css/transition-box-shadows/) about it with some sample code to see the difference.
- For the meta information like the Author, Publish Date, Event Time/Location information, etc., I would consider using a [definition list](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl) so you get the semantics of the key/value pairs. I've started doing this more recently in projects, but what you've got is fine, so up to you if you want to change it.
- Is there a focus state for thumbnail images that have links attached to them? The browser default is fine in this case and matches, but wanted to check.

## Nuclear News
- The one that links to an external link is missing the external link icon next to the title.
- These are the only ones where the whole block is a link - is that intentional? It makes sense since the other ones have links within them, but wanted to confirm!

## Events
- Is the register button supposed to go under the description specifically? Or just under the calendar/description (whichever is longer)?

## Analysis
Looks good!

## Projects
- Can you test one that redirects to an external page to make sure the external link icon is there?

## Programs
Looks good!

## Program Updates
Looks good!

**Comment by HP4k1h5** [comment]
looks great. im trying to understand how `Events` are being sorted. could be because of the test data but currently seeing out-of-order events after the highlighted events.

**Comment by urchykli** [comment]
@jnschrag To answer some of your questions:
- I asked Christina and she confirmed images with links have the default blue outline on focus.
- Yes, news blocks are the only ones where the whole block is a link.
- Event registration buttons go under the calendar/description (whichever is longer). However, the bottom margin on the calendar has a big effect.
- I added a Project post with an external link.  Sorry about that oversight!

**Comment by urchykli** [comment]
@HP4k1h5 The order of the events will happen when that analysis page is put together.  It involves a section for past events and an upcoming events section that displays text if there aren't any upcoming events.

**Pull request merged**